### PR TITLE
Compile-time package parsing

### PR DESCRIPTION
**Pull request by kofigumbs**
You will need to setup travis in order to get automatic deploys to gh-pages
 - Add `GITHUB_TOKEN` environment variable https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/
 - Add cron triggers for nightly builds

Trigger a local build with `make setup build`. For the first time, I would delete `./elm-stuff` then run `make install`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
   "author": "",
-  "license": "ISC"
+  "license": "BSD-3-Clause",
+  "bugs": {
+    "url": "https://github.com/klaftertief/elm-search/issues"
+  },
+  "homepage": "https://github.com/klaftertief/elm-search#readme",
+  "devDependencies": {
+    "elm": "^0.18.0",
+    "elm-test": "^0.18.9",
```
Comment (klaftertief) [inline package.json:17]: There are no tests yet, but I'll keep it as a reminder to add some :-)

File: src/scripts/Blacklist.elm [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module Blacklist exposing (contains, length)
+
+
+length : Int
+length =
+    2
+
+
+contains : { a | user : String, name : String } -> Bool
```
Comment (klaftertief) [inline src/scripts/Blacklist.elm:9]: Why are those packages blacklisted? Renamed use/project names?

File: src/scripts/Blacklist.elm [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module Blacklist exposing (contains, length)
+
+
+length : Int
```
Comment (klaftertief) [inline src/scripts/Blacklist.elm:4]: Where is this needed?

File: src/scripts/Docs/Package.elm [hunk 4]
```diff
@@ -1,42 +1,91 @@
...
+    , modules : List Module
+    }
+
+
+type alias Metadata =
     { user : String
     , name : String
-    , version : Version
-    , modules : List Module
+    , version : String
```
Comment (klaftertief) [inline src/scripts/Docs/Package.elm:26]: Why is this a string now? I guess that we don't need to parse into some fancy record just to turn it into a string afterwards. (This code was mostly taken from the package website where more context is needed. I had some plans to use such data at some point, but not at the moment. So everything is ok.)

File: src/scripts/Web.elm [hunk 5]
```diff
@@ -1,19 +1,146 @@
...
+import Search.Model as Search
+import Search.Update as Search
+import Search.View as Search
 
 
-main : Program Flags Model Msg
-main =
+program : List Package -> Program Flags Model Msg
+program packages =
     Html.programWithFlags
```
Comment (klaftertief) [inline src/scripts/Web.elm:15]: This should become a `Navigation.program` at some point, but not in this PR.

File: Makefile [hunk 6]
```diff
@@ -1,44 +1,47 @@
...
+install: ## Installs dependencies
+	npm install
+	node_modules/.bin/elm-package install --yes
+	[ -f elm-stuff/packages/elm-tools/documentation/*/src/Elm/Documentation/Type.elm.orig ] || \
+		patch --backup elm-stuff/packages/elm-tools/documentation/*/src/Elm/Documentation/Type.elm documentation-empty-record.patch
 
-install: $(INSTALL_TARGETS) ## Installs prerequisites and generates file/folder structure
+installdev: install $(INSTALL_TARGETS) ## Installs prerequisites with development server tools and generates file/folder structure
 
 server: ## Runs a local server for development
```
Comment (klaftertief) [inline Makefile:46]: Did you use the dev server setup? I'll probably just remove it.

**Review by klaftertief — APPROVED** [review]
I just had some more anecdotal comments, so I'm merging here and now and try to use the juice to pick some lower hanging fruits.

Will announce it on Monday then.

Many many thanks again.

File: src/scripts/Blacklist.elm (hunk 3 above)
Comment (kofigumbs) [reply src/scripts/Blacklist.elm:4]: Artifact from an old architecture - safe to delete!

File: src/scripts/Docs/Package.elm (hunk 4 above)
Comment (kofigumbs) [reply src/scripts/Docs/Package.elm:26]: Yep, that's basically what I was thinking

File: src/scripts/Blacklist.elm (hunk 2 above)
Comment (kofigumbs) [reply src/scripts/Blacklist.elm:9]: `tea-component` was manually deleted ([source](https://groups.google.com/d/msg/elm-dev/iqErmKHnLDY/DXPOSOwsAQAJ)). I don't know the story behind `elm-i18next`, but if you check Elm Package, there are no published versions. I'm guessing it was manually deleted as well.

**Pull request merged**